### PR TITLE
fix and re-organize docs/

### PR DESCRIPTION
**Pull request by tkdchen**
- re-organize format of *.rst and *.md in docs/
- fix error link of vagrant-adbinfo in using.rst
- remove trailing whitespaces

This change is intended to make document more readable even in a text
editor.

### CODE REVIEW

File: docs/architecture.rst [hunk 1]
```diff
@@ -36,31 +54,59 @@ These boxes contains the following components from the latest CentOS Core:
     * nfs-utils
     * PyYAML
     * libyaml-devel
-  * atomic - The `Atomic CLI <https://github.com/projectatomic/atomic>`_
+  * atomic - The `Atomic CLI`_
```
Comment (bexelbie) [inline docs/architecture.rst]: Why was this url dropped?

File: docs/architecture.rst [hunk 2]
```diff
@@ -36,31 +54,59 @@ These boxes contains the following components from the latest CentOS Core:
...
+
+  * OpenShift - A PAAS environment for source-to-image container execution in a
+    devops pattern
+
+  * `Atomic App`_ - The `Nulecule`_ container application specification
+    reference implementation
+
+  * `atomic-reactor`_ - a python based container builder
+
+  * `DevAssistant`_ - used to provide templates for `Dockerfiles`_ and `Nulecule
```
Comment (bexelbie) [inline docs/architecture.rst:93]: Unrelated directly to your edits directly, we need to move DA into the "future" as that is not yet ready or containerized, afaik.

Comment (bexelbie) [inline docs/architecture.rst:93]: This doesn't need to be done in this PR.

**Comment by bexelbie** [comment]
LGTM

**Comment by LalatenduMohanty** [comment]
LGTM

**Pull request merged**